src/firefly/js/drawingLayers/FootprintTool.js

```diff
@@ -0,0 +1,538 @@
+    dl.regions = get(initPayload, 'regions', null);
+    dl.fpInfo = get(initPayload, 'fpInfo', {});
+    return dl;
+}
+
+
+/**
+ * reducer for MarkerTool layer
+ * @param drawLayer
+ * @param action
+ * @returns {*}
+ */
+function getLayerChanges(drawLayer, action) {
+    var {drawLayerId} = action.payload;
+
+    if (!drawLayerId || drawLayerId !== drawLayer.drawLayerId) return null;
+    var dd = Object.assign({}, drawLayer.drawData);
+
+    switch (action.type) {
+
+        case DrawLayerCntlr.FOOTPRINT_CREATE:
+        case DrawLayerCntlr.FOOTPRINT_START:
+        case DrawLayerCntlr.FOOTPRINT_MOVE:
+        case DrawLayerCntlr.FOOTPRINT_END:
+            var data = get(dd, [DataTypes.DATA, '0']);
+            var {text, textLoc} = data || {};
+            var crtFpObj = data || null;
+
+            return createFootprintObjs(action, text, textLoc, crtFpObj);
+
+        case DrawLayerCntlr.MODIFY_CUSTOM_FIELD:
+            var obj = get(dd, [DataTypes.DATA, '0']);
+            if (!obj) return null;
+            var {fpText, fpTextLoc, angleDeg} = action.payload.changes;
+            var {plotIdAry} = action.payload;
+
+
+            if (!isNil(angleDeg)) {
+                return updateFootprintAngle(angleDeg, obj, plotIdAry[0]);
+            } else {
+                return updateFootprintText(fpText, fpTextLoc, obj);
+            }
+    }
+}
+
+function getCursor(plotView, screenPt) {
+    var  cursor = '';
+    var dlAry = dlRoot().drawLayerAry.filter( (dl) => {
+        return (dl.drawLayerTypeId === TYPE_ID) && (get(dl, 'visiblePlotIdAry').includes(plotView.plotId));
+    });
+
+    if (!screenPt) {
+        //alert('null screenpt');
+        return cursor;
+    }
+    const plot= primePlot(plotView);
+    var   cc= CsysConverter.make(plot);
+
+    dlAry.find( (dl) => {
+        var drawObj = get(dl, ['drawData', 'data', '0']);
+        var idx = findClosestIndex(screenPt, drawObj, cc).index;
+
+        if (idx >= 0 && idx <= drawObj.outlineIndex) {
+            cursor = 'pointer';
+            return true;
+        } else if (idx === drawObj.rotateIndex) {
+            cursor = 'alias';
+            return true;
+        } else {
+            return false;
+        }
+    });
+    return cursor;
+}
+
+/**
+ * update text in marker object
+ * @param text
+ * @param textLoc
+ * @param footprintDrawObj
+ * @returns {*}
+ */
+function updateFootprintText(text, textLoc, footprintDrawObj) {
+    var textUpdatedObj = updateFootprintDrawobjText(footprintDrawObj, text, textLoc);
+
+    return textUpdatedObj? {drawData: {data: [textUpdatedObj]}} : {};
+}
+
+/**
+ * set footprint rotate angle
+ * @param angleDegStr
+ * @param footprintDrawObj
+ * @param plotId
+ * @returns {*}
+ */
+function updateFootprintAngle(angleDegStr, footprintDrawObj, plotId) {
+    const plot = primePlot(visRoot(), plotId);
+    var cc = CsysConverter.make(plot);
+    var angleDeg;
+
+    angleDeg = angleDegStr ? parseFloat(angleDegStr) : 0.0;
```

Comment:
after reading javascript docs, it seems that parseFloat is the highest precision available, the loss of precision must be somewhere else where we transform angles and distances.